disnake/ext/commands/base_core.py

```diff
@@ -627,8 +627,8 @@ async def can_run(self, inter: ApplicationCommandInteraction) -> bool:
             if not predicates:
                 # since we have no checks, then we just return True.
                 return True
-
-            return await async_all(predicate(inter) for predicate in predicates)  # type: ignore
+            else:
+                return await async_all(predicate(inter) for predicate in predicates)  # type: ignore
```

Comment:
i'll disable it for now because you're right. But we should consider using it in the future. IIRC there's a way that python can be hiding tracebacks with the `finally` block and returning within the `try:`

Reply:
https://github.com/DisnakeDev/disnake/pull/935/commits/9d129c0c55bf70a38e9027238a59f303af968eeb